debugger/src/tpage.c

```diff
@@ -161,23 +166,41 @@ void tpage_pack_widgets(gboolean tabbed)
 	{
 		GtkWidget *hbox, *rbox, *lbox;
 
+#if GTK_CHECK_VERSION(3, 0, 0)
+		root = gtk_box_new(GTK_ORIENTATION_VERTICAL, SPACING);
+#else
 		root = gtk_vbox_new(FALSE, SPACING);
+#endif
 		gtk_container_set_border_width(GTK_CONTAINER(root), ROOT_BORDER_WIDTH);
 	
 		/* filename */
+#if GTK_CHECK_VERSION(3, 0, 0)
+		hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, SPACING);
+#else
 		hbox = gtk_hbox_new(FALSE, SPACING);
+#endif
 		gtk_box_pack_start(GTK_BOX(root), hbox, FALSE, FALSE, 0);
 		gtk_box_pack_start(GTK_BOX(hbox), target_label, FALSE, FALSE, 0);
 		gtk_box_pack_start(GTK_BOX(hbox), target_name, TRUE, TRUE, 0);
 		gtk_box_pack_start(GTK_BOX(hbox), target_button_browse, FALSE, FALSE, 0);
 		
 		/* lower hbox */
+#if GTK_CHECK_VERSION(3, 0, 0)
+		hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, SPACING);
+		gtk_box_set_homogeneous(GTK_BOX(hbox), TRUE);
```

Comment:
Looks like ```gtk_box_set_homogeneous()``` could be used in both gtk versions?

Reply:
Maybe it could, but `gtk_box_set_homogeneous()` is used because of the missing parameter in `gtk_box_new()`. Using it in the GTK2 version would be redundant.

Reply:
Ok, then let's keep it as-is.